src/python/pants/task/simple_codegen_task.py

```diff
@@ -210,23 +211,45 @@ def synthetic_target_dir(self, target, target_workdir):
     """
     return target_workdir
 
-  def _inject_synthetic_target(self, target, target_workdir):
+  def _create_sources_with_fingerprint(self, target_workdir, fingerprint, files):
+    """Create an EagerFilesetWithSpec to pass to the sources argument for synthetic target injection.
+
+    We are creating and passing an EagerFilesetWithSpec to the synthetic target injection in the
+    hopes that it will save the time of having to refingerprint the sources.
+
+    :param target_workdir: The directory containing the generated code for the target.
+    :param fingerprint: the fingerprint of the VersionedTarget with which the EagerFilesetWithSpec
+           will be created.
+    :param files: a list of exact paths to generated sources.
+    """
+    results_dir_relpath = os.path.relpath(target_workdir, get_buildroot())
+    filespec = FilesetRelPathWrapper.to_filespec(
+      [os.path.join(results_dir_relpath, file) for file in files])
+    return EagerFilesetWithSpec(results_dir_relpath, filespec=filespec,
+      files=files, files_hash='{}.{}'.format(fingerprint.id, fingerprint.hash))
+
+
+  def _inject_synthetic_target(self, target, target_workdir, fingerprint=None):
```

Comment:
This method isn't marked `:API: public`, so you can freely add/remove required parameters: would make `fingerprint` required here.